core/tee/fs_htree.c

```diff
@@ -495,11 +497,12 @@ static TEE_Result authenc_init(void **ctx_ret, TEE_OperationMode mode,
 					TEE_FS_HTREE_IV_SIZE);
 
 exit:
-	if (res == TEE_SUCCESS)
+	if (res == TEE_SUCCESS) {
 		*ctx_ret = ctx;
-	else
+	} else {
 		crypto_authenc_final(ctx, alg);
-
+		crypto_authenc_free_ctx(ctx, alg);
```

Comment:
`crypto_authenc_final()` should be done only if `crypto_authenc_init()` succeeded.
`crypto_authenc_free_ctx()` should be called only if `crypto_authenc_alloc()` succeeded (that is the actual fix for the memory leak)
These two changes should go in there own patch "core: FS: htree: fix potential memory leak".